### ISSUE TITLE
Package porting: galera

### ISSUE THREAD

**Issue by jackiehjm** [Wind-River/meta-akraino#17]
It is required in:
https://gerrit.akraino.org/r/gitweb?p=ta/manifest.git;a=blob_plain;f=packages.yaml;hb=HEAD

In build185:
[newadmin@controller-1 ~]$ rpm -ql galera-25.3.16-3.el7.x86_64
/etc/sysconfig/garb
/usr/lib/systemd/system/garbd.service
/usr/lib64/galera
/usr/lib64/galera/libgalera_smm.so
/usr/sbin/garbd
/usr/sbin/garbd-wrapper
/usr/share/doc/galera
/usr/share/doc/galera/COPYING
/usr/share/doc/galera/LICENSE.asio
/usr/share/doc/galera/LICENSE.chromium
/usr/share/doc/galera/LICENSE.crc32
/usr/share/doc/galera/README
/usr/share/doc/galera/README-MySQL

**Comment by jackiehjm** [comment]
https://github.com/codership/galera

**Comment by gaolitao** [comment]
This pkg

- depends on the scons for building. And the source code from **two** repos

```bash

git@github.com:codership/galera.git
commit of 5c765eb1f5734856f6a6f4a705c785941ae4f476    : this if for release 25.3.16

git@github.com:codership/wsrep-API.git
commit of eab2d5d5a31672c0b7d116ef1629ff18392fd7d0    : this is for v25

```

- files (this list is summarized from the comparison between the scons build result and CentOS install)

```

/etc/sysconfig/garb  (from ${S}/garb/files/garb.cnf)
/usr/bin/garb-systemd (from ${S}/garb/files/garb-systemd)
/usr/lib/systemd/system/garbd.service (from ${S}/garb/files/garb.service)
/usr/lib64/galera/libgalera_smm.so (from ${S}/libgalera_smm.so)
/usr/sbin/garbd (from ${S}/garb/garbd)
```

- requires

```

/usr/bin/sh | -
-- | --
libboost_program_options.so.1.53.0()(64bit) | -
libc.so.6(GLIBC_2.14)(64bit) | -
libcrypto.so.10()(64bit) | -
libcrypto.so.10(libcrypto.so.10)(64bit) | -
libgcc_s.so.1()(64bit) | -
libgcc_s.so.1(GCC_3.0)(64bit) | -
libm.so.6()(64bit) | -
libm.so.6(GLIBC_2.2.5)(64bit) | -
libpthread.so.0()(64bit) | -
libpthread.so.0(GLIBC_2.2.5)(64bit) | -
libpthread.so.0(GLIBC_2.3.2)(64bit) | -
librt.so.1()(64bit) | -
librt.so.1(GLIBC_2.2.5)(64bit) | -
libssl.so.10()(64bit) | -
libssl.so.10(libssl.so.10)(64bit) | -
libstdc++.so.6()(64bit) | -
libstdc++.so.6(CXXABI_1.3)(64bit) | -
libstdc++.so.6(GLIBCXX_3.4)(64bit) | -
libstdc++.so.6(GLIBCXX_3.4.10)(64bit) | -
libstdc++.so.6(GLIBCXX_3.4.11)(64bit) | -
libstdc++.so.6(GLIBCXX_3.4.15)(64bit) | -
libstdc++.so.6(GLIBCXX_3.4.9)(64bit) | -
nmap-ncat | -
rtld(GNU_HASH) | -
systemd



```

**Comment by gaolitao** [comment]
In current yocto code base, OpenSSL 1.1.0 has been used, but the issue which tracks "Galera incompatible with OpenSSL 1.1.0" (https://github.com/codership/galera/issues/407) is being observed in galera-25.3.16.

According to the code inspection, OpenSSL 1.1 compatibility has been resolved in galera-25.3.29.

Two options:
- disable ssl usage in galera
- use version 25.3.29 instead of 25.3.16.

For now, i will go for 2nd option.

```
In file included from /home/lgao2/akraino/prj_wrl1018_akraino/tmp-glibc/work/corei7-64-oe-linux/galera/25.3.16-r0/recipe-sysroot/usr/include/openssl/bio.h:20,
                 from /home/lgao2/akraino/prj_wrl1018_akraino/tmp-glibc/work/corei7-64-oe-linux/galera/25.3.16-r0/recipe-sysroot/usr/include/openssl/conf.h:13,
                 from asio/asio/ssl/detail/openssl_types.hpp:19,
                 from asio/asio/ssl/context_base.hpp:20,
                 from asio/asio/ssl/basic_context.hpp:25,
                 from asio/asio/ssl.hpp:18,
                 from .sconf_temp/conftest_12.cpp:2:
asio/asio/ssl/detail/openssl_init.hpp: In constructor 'asio::ssl::detail::openssl_init<Do_Init>::do_init::do_init()':
asio/asio/ssl/detail/openssl_init.hpp:51:27: error: expected id-expression before '(' token
         mutexes_.resize(::CRYPTO_num_locks());
                           ^~~~~~~~~~~~~~~~
In file included from asio/asio/ssl/detail/openssl_context_service.hpp:26,
                 from asio/asio/ssl/context_service.hpp:25,
                 from asio/asio/ssl/context.hpp:21,
                 from asio/asio/ssl.hpp:19,
                 from .sconf_temp/conftest_12.cpp:2:
asio/asio/ssl/detail/openssl_init.hpp:54:70: error: expected id-expression before ';' token
         ::CRYPTO_set_locking_callback(&do_init::openssl_locking_func);
                                                                      ^
asio/asio/ssl/detail/openssl_init.hpp:55:60: error: expected id-expression before ';' token

```
